lib/chef/provider/remote_file/fetcher.rb

```diff
@@ -24,6 +24,9 @@ class Fetcher
 
         def self.for_resource(uri, new_resource, current_resource)
           if network_share?(uri)
+            if !Chef::Platform.windows?
+              raise Exceptions::UnsupportedPlatform, "Fetching the file on a network share is supported only on the Windows platform. Please change your source: #{uri}"
+            end
```

Comment:
Actually looking closer I'm wrong. This only makes sence on Windows unless using a smbclient which we dont do here.